test/unit/event.js

```diff
@@ -1289,6 +1289,27 @@ QUnit.test( "Delegated events in SVG (#10791; #13180)", function( assert ) {
 	jQuery( "#qunit-fixture" ).off( "click" );
 } );
 
+QUnit.test( "Delegated events with malformed selectors (#3071)", function( assert ) {
+	assert.expect( 2 );
+	try {
+		jQuery( "#qunit-fixture" ).on( "click", "div:not", function () { } );
+		assert.ok( false, "malformed selector throws on attach" );
+	}
+	catch ( e ) {
+		assert.ok( true, "malformed selector throws on attach" );
```

Comment:
That would leave this asymmetric with the below call (I don't see a `notThrows` assert method). I you think that doesn't matter, I can change that.